src/game-logic/cards-collection.ts

```diff
@@ -269,6 +270,27 @@ export const cardModels: any = {
     "Invoke by sacrifying a card from your field. Then withdraw a card from your opponent field.",
     simpleAttackInfo
   ),
+  Raid: new CardModel(
+    "Raid",
+    raid,
+    0,
+    0,
+    Color.Red,
+    (card: Card) => {
+      if (
+        card.duel.cards[card.playerId][Zone.Deck].length >= 3 &&
```

Comment:
Check the conditions separately.
Give the user a proper alert message. Then return to exit the method.
If neither of the contitions are true, then go ahead and execute the logic of this method